bindings/python/crocoddyl/utils/__init__.py

```diff
@@ -1047,8 +1049,8 @@ def tryStep(self, stepLength=1):
 
     def solve(self, init_xs=[], init_us=[], maxiter=100, isFeasible=False, regInit=None):
         self.setCandidate(init_xs, init_us, isFeasible)
-        self.x_reg = regInit if regInit is not None else self.regMin
-        self.u_reg = regInit if regInit is not None else self.regMin
+        self.x_reg = regInit if regInit is not None else 1e-9
+        self.u_reg = regInit if regInit is not None else 1e-9
```

Comment:
This indeed is not a change. The previous value was `1e-9`, but now the minimum regularization value is not `1e-9`.
So, the short answer is `I didn't want to change this, otherwise, it will affect the log files`.

But actually, there is a better explanation.
Using the minimum regularization implies
   `the default regularization believes that starting from a Gauss-Newton direction is better`.
Instead, using `1e-9` implies that
   `the default regularization assumes that it is more robust to start somewhere between Gauss-Newton to steepest descent, however, it is good to be closed to Gauss-Newton one`.

This choice was designed by Nicolas, however, I am inclined to keep this strategy.

Other comments, note that it is important to keep the same values in these Python derived classes and c++ code for unit-testing purpose.

I consider this conversation as resolved, however, feel free to open it if this need further discussion.